### PR TITLE
PCHR-2561: Responsive My Leave Page

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR(and https://github.com/compucorp/civihr-employee-portal-theme/pull/217) responsive design has been implemented to the My Leave Page of SSP.

### iPhone 6 Landscape
my-leave-iphone-land

### iPhone 6 Portrait
my-leave-iphone-port

### iPad Landscape
my-leave ipad -land

### iPad Portrait
my-leave-ipad-port

### Desktop
my-leave

## Technical Details

1. All tables inside `.chr_leave-report` are set with `table-layout: fixed;`. This is done to fix the width of all tables.
2. Width for all columns of the table set to `px`, this is done so that all the tables have constant width and does not depend on the content inside.
3. `chr_leave-report__heading panel-heading` is moved outside of `panel panel-default`, and a new div with`chr_leave-report__table-container` class added outside of all the tables, so that they can be scrolled together.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss [hunk 1]
```diff
@@ -8,6 +8,18 @@
       background: #e6e6e6;
     }
   }
+
+  table {
+    table-layout: fixed;
+
+    thead th,
+    tbody td {
+      overflow:hidden;
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss]: space `overflow: hidden;`

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss [hunk 2]
```diff
@@ -8,6 +8,18 @@
...
+      padding: 8px 5px;
+      white-space: normal !important;
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss]: why `!important`? can it be avoided? (if not, that's ok)

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 3]
```diff
@@ -53,48 +54,48 @@
...
-                      class="table-nested table-responsive"
+                      class="table-nested"
                       ng-show="report.sections.entitlements.open"
                       ng-switch="!!report.sections.entitlements.data.length">
                       <table class="table table-condensed" ng-switch-when="false">
                         <tbody>
-                          <tr>
-                            <td align="center">No entitlement breakdown data available</td>
-                          </tr>
+                        <tr>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: fix wrong indentation:

```pug
tbody
  tr
    td
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 4]
```diff
@@ -53,48 +54,48 @@
...
-                            <th class="chr_leave-report__table__heading--dates">Dates</th>
-                            <th class="chr_leave-report__table__heading--status">Status</th>
-                            <th
-                              class="chr_leave-report__table__heading--type"
-                              ng-repeat="absenceType in report.absenceTypesFiltered track by $index">
-                              {{absenceType.title}}
-                            </th>
-                            <th class="chr_leave-report__table__heading--action"></th>
-                          </tr>
+                        <tr>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: fix wrong indentation (and all others please)

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss (hunk 2 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-report.scss]: I double checked, and this whole css style is not required.

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 5]
```diff
@@ -18,57 +17,59 @@
...
-                        <tbody>
-                          <tr>
-                            <td align="center">No entitlement breakdown data available</td>
-                          </tr>
-                        </tbody>
-                      </table>
-                      <table class="table table-condensed" ng-switch-when="true">
-                        <thead>
-                          <tr>
+    <div class="chr_leave-report__table-container">
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: why is the previous element padding bigger than this one if they are supposed to be at the same padding level?

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 6]
```diff
@@ -79,14 +80,14 @@
                             </th>
                             <th class="chr_leave-report__table__heading--action"></th>
                           </tr>
-                        </thead>
-                        <tbody>
-                          <tr ng-repeat="entry in report.sections.entitlements.data track by $index">
+                          </thead>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: This one didn't need padding. Needs to be removed.

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 7]
```diff
@@ -79,14 +80,14 @@
...
+                          <tbody>
+                            <tr ng-repeat="entry in report.sections.entitlements.data track by $index">
                             <td class="chr_leave-report__label">
                               {{entry.type.label}}
                               <span ng-if="entry.expiry_date">
-                                (expires {{entry.expiry_date | date:report.dateFormat}})
-                              </span>
+                                  (expires {{entry.expiry_date | date:report.dateFormat}})
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: This line didn't need the extra padding.

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 8]
```diff
@@ -95,46 +96,46 @@
...
-                <td class="table-nested-cell" colspan="9">
-                  <crm-loading show="!report.sections.holidays.loading">
-                    <div
-                      class="table-nested table-responsive"
-                      ng-show="report.sections.holidays.open"
-                      ng-switch="!!report.sections.holidays.data.length">
-                      <table class="table table-condensed" ng-switch-when="false">
-                        <tbody>
-                          <tr>
+                          </tbody>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: Didn't need padding

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 9]
```diff
@@ -271,14 +272,14 @@
...
-                          <tr ng-repeat="entry in report.sections.expired.data track by $index">
+                          </thead>
+                          <tbody>
+                            <tr ng-repeat="entry in report.sections.expired.data track by $index">
                             <td class="chr_leave-report__label">
                               {{entry.type.label}}
                               <span ng-if="entry.expiry_date">
-                                (expired {{entry.expiry_date | date:report.dateFormat}})
-                              </span>
+                                  (expired {{entry.expiry_date | date:report.dateFormat}})
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: Didn't need the extra padding

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 10]
```diff
@@ -287,66 +288,66 @@
                             </td>
                             <td></td>
                           </tr>
-                        </tbody>
-                      </table>
-                    </div>
-                  </crm-loading>
+                          </tbody>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: The padding here is wrong

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 11]
```diff
@@ -357,14 +358,14 @@
...
-                        <tr ng-repeat="request in report.sections.pending.data track by $index">
+                        </thead>
+                        <tbody>
+                          <tr ng-repeat="request in report.sections.pending.data track by $index">
                           <td class="chr_leave-report__label">
                             {{report.absenceTypes[request.type_id].title}}
                             <span ng-if="request.toil_expiry_date">
-                              (expires {{request.toil_expiry_date | date:report.dateFormat}})
-                            </span>
+                                (expires {{request.toil_expiry_date | date:report.dateFormat}})
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: didn't need the extra padding

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 12]
```diff
@@ -444,14 +450,14 @@
                           </th>
                           <th class="chr_leave-report__table__heading--action"></th>
                         </tr>
-                      </thead>
-                      <tbody>
-                        <tr ng-repeat="request in report.sections.other.data track by $index">
+                        </thead>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: didn't need padding

File: uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html [hunk 13]
```diff
@@ -469,15 +475,16 @@
...
-                    </table>
-                  </div>
-                </crm-loading>
-              </td>
-            </tr>
-          </tbody>
-        </table>
-      </div>
-    </section>
+                        </tbody>
```
Comment (reneolivo) [inline uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/staff-leave-report.html]: this padding was fine.

**Pull request merged**